### ISSUE TITLE
Support for `ZodReadonly` and `.readonly()`

### ISSUE THREAD

**Issue by BP18** [asteasolutions/zod-to-openapi#173]
Zod [Release v3.22.0](https://github.com/colinhacks/zod/releases/tag/v3.22.0) (2023-08-14) states:
> This release introduces `ZodReadonly` and the `.readonly()` method on `ZodType`.

Related docs can be found here: https://zod.dev/?id=readonly

Currently (version `v5.5.0`) `zod-to-openapi` throws an error for both the following examples:

**1.** `z.object({ foo: z.string() }).readonly()`:
```
UnknownZodTypeError {
  message: 'Unknown zod object type, please specify `type` and other OpenAPI props using `ZodSchema.openapi`.',
  data: {
    currentSchema: {
      innerType: ZodObject { ... },
      typeName: 'ZodReadonly'
    }
  }
}
```

**2.** `z.string().array().readonly()`:
```
UnknownZodTypeError {
  message: 'Unknown zod object type, please specify `type` and other OpenAPI props using `ZodSchema.openapi`.',
  data: {
    currentSchema: {
      innerType: ZodArray { ... },
      typeName: 'ZodReadonly'
    }
  }
}
```

`zod-to-openapi` should likely handle these examples without error? Inferring semantics around OpenAPI's [Read-Only and Write-Only Properties](https://swagger.io/docs/specification/data-models/data-types/) is probably a more complex future opportunity?

**Comment by AGalabov** [comment]
@BP18 thank you for reporting this,. It most definitely makes sense. As for read-only and write-only from OpenAPI - from what I read I do not think they have the same semantics as the `.readonly` `zod` modifier. So I think I'd rather just "unwrap" it and generate docs based on the inner schema

**Comment by AGalabov** [comment]
@BP18 this should now be available as part of the [v6.1.0](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v6.1.0) release

**Comment by BP18** [comment]
> @BP18 this should now be available as part of the [v6.1.0](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v6.1.0) release

Thank you!
